src/firefly/js/ui/NaifidPanel.jsx

```diff
@@ -0,0 +1,200 @@
+            popStyle={popStyle}
+            message={message}
+            value={value}
+            valueOnSuggestion={getNaifidValue((selectedSugg) => updateFeedback(selectedSugg, fieldKey, groupKey, valid))}
+            getSuggestions={this.getSuggestions}
+            renderSuggestion={renderSuggestion}
+        />);
+        let naifidFeedback = (<TargetFeedback {...{feedback}} {...{feedbackStyle}}/>);
+
+        return (
+            <div>
+                <div>{positionField}</div>
+                <div>{naifidFeedback}</div>
+            </div>
+        );
+    }
+
+}
+
+NaifidPanelView.propTypes = {
+    fieldKey : PropTypes.string,
+    groupKey : PropTypes.string,
+    label : PropTypes.string,
+    valid   : PropTypes.bool.isRequired,
+    showHelp   : PropTypes.bool.isRequired,
+    feedback: PropTypes.string.isRequired,
+    message: PropTypes.string.isRequired,
+    value : PropTypes.string.isRequired,
+    labelWidth : PropTypes.number,
+    popStyle : PropTypes.object, //style for the suggestion box popup list
+    onUnmountCB : PropTypes.func,
+    feedbackStyle: PropTypes.object
+};
+
+
+
+function didUnmount(fieldKey,groupKey, props) {
+    const wp= parseWorldPt(FieldGroupUtils.getFldValue(FieldGroupUtils.getGroupFields(groupKey),fieldKey));
+
+    if (props.nullAllowed && !wp) {
+        dispatchActiveTarget(null);
+    }
+    else if (isValidPoint(wp)) {
+        dispatchActiveTarget(wp);
+    }
+}
+
+
+
+function getProps(params) {
+    let feedback= params.feedback|| '';
+    let value= params.displayValue;
+    let showHelp= get(params,'showHelp', true);
+    const wpStr= params.value;
+    const wp= parseWorldPt(wpStr);
+
+    if (isValidPoint(wp) && !value) {
+        feedback= getFeedback(wp);
+        value= wp.objName || formatPosForTextField(wp);
+        showHelp= false;
+    }
+
+    return Object.assign({}, params,
+        {
+            visible: true,
+            label: params.label || LABEL_DEFAULT,
+            tooltip: 'Enter a target',
+            value,
+            feedback,
+            showHelp,
+            nullAllowed:params.nullAllowed,
+            onUnmountCB: didUnmount
+        });
+}
+
+
+function updateFeedback(value, fieldKey, groupKey, valid){
+   const payload={
+       fieldKey: fieldKey,
+       groupKey: groupKey,
+       feedback: value,
+       valid: valid
+   };
+   dispatchValueChange(payload);
+}
+
+
+const connectorDefaultProps = {
+    fieldKey : 'UserTargetWorldPt',
+    initialState  : {
+        fieldKey : 'UserTargetWorldPt'
+    }
+};
+
+function replaceValue(v,props) {
+    const t= getActiveTarget();
+    // if (props.nullAllowed && !v) return v;
+    let retVal= v;
+    if (t && t.worldPt) {
+        if (get(t,'worldPt')) retVal= t.worldPt.toString();
+    }
+    return retVal;
+}
+
+function getNaifidValue(onCallBack) {
+    return (val, str) => {
+        if (! str) return;
+
+        const naifId = (str.match(/Naifid:.*/))[0].split('Naifid:').pop();
```

Comment:
This is error-prone.  When match fail, `[0]` will cause TypeError.
Should replace with this:
`const [,naifId] = str.match(/Naifid:(.*)/) || [];`

Reply:
Done!
Thanks :)